src/Cech_complex/include/gudhi/Cech_complex_blocker.h

```diff
@@ -90,25 +91,28 @@ class Cech_blocker {
                 std::clog << "#(" << vertex << ")#";
 #endif  // DEBUG_TRACES
             }
-            sph = get_sphere(points.cbegin(), points.cend());
             // Put edge sphere in cache
-            sc_ptr_->assign_key(face_opposite_vertex.first, cc_ptr_->get_cache().size());
-            cc_ptr_->get_cache().push_back(sph);
+            sph_key = cc_ptr_->get_cache().size();
+            sc_ptr_->assign_key(face_opposite_vertex.first, sph_key);
+            cc_ptr_->get_cache().push_back(get_sphere(points.cbegin(), points.cend()));
             // Clear face points
             points.clear();
         }
         // Check if the minimal enclosing ball of current face contains the extra point/opposite vertex
+        Sphere const& sph = cc_ptr_->get_cache()[sph_key];
         if (kernel_.squared_distance_d_object()(sph.first, cc_ptr_->get_point(face_opposite_vertex.second)) <= sph.second) {
-#ifdef DEBUG_TRACES
-            std::clog << "center: " << sph.first << ", radius: " <<  radius << std::endl;
-#endif  // DEBUG_TRACES
             is_min_enclos_ball = true;
+            sc_ptr_->assign_key(sh, sph_key);
 #if CGAL_VERSION_NR >= 1050000000
             if(exact_) CGAL::exact(sph.second);
 #endif
-            radius = std::sqrt(cast_to_fv(sph.second));
-            sc_ptr_->assign_key(sh, cc_ptr_->get_cache().size());
-            cc_ptr_->get_cache().push_back(sph);
+            if(k != sc_ptr_->null_key())
+                radius = sc_ptr_->filtration(face_opposite_vertex.first);
```

Comment:
Can you clarify why we need this test before taking the filtration value?
IIRC, k is null_key when we first look at an edge e as a boundary of a triangle, because we haven't added it to the cache yet, but we do have its filtration value in the simplex tree already.

Reply:
Indeed.